solutions/gold/cf-1187E.mdx

```diff
@@ -0,0 +1,172 @@
+---
+id: cf-1187E
+source: CF
+title: Tree Painting
+author: Yash Singh
+---
+
+[Official Editorial](https://codeforces.com/blog/entry/68111)
+
+## Explanation
+
+We notice that choosing a vertice splits the current connected component into
+the childrens' subtrees and the rest of the tree excluding the current subtree.
+Another observation we can make is that choosing a leaf node is more optimal
+than choosing anything in the middle because that way we have the entire size of
+the rest of the tree as a part of the leftover connected component. This hints
+for a DP on Trees solution, we choose one leaf node, work our way up and work
+our way down for each of the siblings. For example, for the following tree:
+
+![Tree Visualization](cf-1187E/tree.png)
+
+If we select the 4th node as our starting point, then the process would look
+like the following:
+
+1. Work our way up to 3
```

Comment:
again, what does "work our way up" mean

Reply:
Work our way up means to start from the leaf node and go through each of the ancestors.

Reply:
and...what are you doing when going through the ancestors?